Mathlib/RingTheory/HahnSeries/Binomial.lean

```diff
@@ -0,0 +1,170 @@
+terms give a formal expansion of `x^r` as `(1 + (x-1))^r`. -/
+def binomialFamily (x : HahnSeries Γ A) (r : R) :
+    SummableFamily Γ A ℕ :=
+  powerSeriesFamily (x - 1) (PowerSeries.binomialSeries A r)
+
+@[simp]
+theorem binomialFamily_apply {x : HahnSeries Γ A} (hx : 0 < (x - 1).orderTop) (r : R) (n : ℕ) :
+    binomialFamily x r n = Ring.choose r n • (x - 1) ^ n := by
+  simp [hx, binomialFamily]
+
+@[simp]
+theorem binomialFamily_apply_of_orderTop_nonpos {x : HahnSeries Γ A} (hx : ¬ 0 < (x - 1).orderTop)
+    (r : R) (n : ℕ) :
+    binomialFamily x r n = 0 ^ n := by
+  rw [binomialFamily, powerSeriesFamily_of_not_orderTop_pos hx]
+  by_cases hn : n = 0 <;> simp [hn]
+
+theorem binomialFamily_orderTop_pos {x : HahnSeries Γ A} (hx : 0 < (x - 1).orderTop) (r : R) {n : ℕ}
+    (hn : 0 < n) :
+    0 < (binomialFamily x r n).orderTop := by
+  simp only [binomialFamily, smulFamily_toFun, PowerSeries.binomialSeries_coeff, powers_toFun, hx,
+    ↓reduceIte, smul_assoc, one_smul]
+  have : n ≠ 0 := by exact Nat.ne_zero_of_lt hn
+  calc
+    0 < n • (x - 1).orderTop := (nsmul_pos_iff (Nat.ne_zero_of_lt hn)).mpr hx
+    n • (x - 1).orderTop ≤ ((x - 1) ^ n).orderTop := orderTop_nsmul_le_orderTop_pow
+    ((x - 1) ^ n).orderTop ≤ ((Ring.choose r n) • ((x - 1) ^ n)).orderTop :=
+      orderTop_le_orderTop_smul (Ring.choose r n) ((x - 1) ^ n)
+
+theorem binomialFamily_mem_support {x : HahnSeries Γ A}
+    (hx : 0 < (x - 1).orderTop) (r : R) (n : ℕ) {g : Γ}
+    (hg : g ∈ (binomialFamily x r n).support) : 0 ≤ g := by
+  by_cases hn : n = 0; · simp_all
+  exact le_of_lt (WithTop.coe_pos.mp (lt_of_lt_of_le (binomialFamily_orderTop_pos hx r
+    (Nat.pos_of_ne_zero hn)) (orderTop_le_of_coeff_ne_zero hg)))
+
+theorem orderTop_hsum_binomialFamily_pos {x : HahnSeries Γ A} (hx : 0 < (x - 1).orderTop)
+    (r : R) : (0 : WithTop Γ) < (SummableFamily.hsum (binomialFamily x r) - 1).orderTop := by
+  obtain (_|_) := subsingleton_or_nontrivial A
+  · simp [Subsingleton.eq_zero ((binomialFamily x r).hsum - 1)]
+  · refine (orderTop_self_sub_one_pos_iff (binomialFamily x r).hsum).mpr ?_
+    constructor
+    · exact hsum_orderTop_of_le (by simp [hx]) (fun b g hg => binomialFamily_mem_support
+        hx r b hg) fun b hb => coeff_eq_zero_of_lt_orderTop <| binomialFamily_orderTop_pos hx r <|
+        Nat.zero_lt_of_ne_zero hb
+    · have : (binomialFamily x r 0).coeff 0 = 1 := by simp [hx]
+      rw [← this]
+      refine hsum_leadingCoeff_of_le (g := 0) (a := 0) (by simp [hx]) ?_ ?_
+      · intro b g' hg'
+        exact binomialFamily_mem_support hx r b hg'
+      · intro b hb
+        exact coeff_eq_zero_of_lt_orderTop <| binomialFamily_orderTop_pos hx r <|
+        Nat.zero_lt_of_ne_zero hb
+
+end SummableFamily
+
+/-- The group of invertible Hahn series close to 1, i.e., those series such that subtracting 1
+  yields a series with strictly positive `orderTop`. -/
+@[simps]
```

Comment:
Do you really want this simps lemma if we already have `mem_orderTopSubOnePos_iff` below?